polyglot.js

```diff
@@ -168,14 +162,28 @@ class PolyGlot {
         select.val(selectedLanguage);
     }
 
-    randomRune() {
-        return String.fromCharCode(Math.floor(Math.random() * 0x51) + 0x16A0)
+    generateRune(string,salt) {
+        var hash = 0;
+        salt = string+salt;
+        for (var i = 0; i < salt.length; i++) {
+            var char = salt.charCodeAt(i);
+            hash = ((hash << 5) - hash) + char;
+            hash = hash & hash;
+        }
```

Comment:
I don't understand this algorithm. I shouldn't need to reverse engineer it to understand what it does. It looks like it's hashing the string with the salt. I would move this into a separate `hashString` function so we know exactly what it does, since it's not part of actually scrambling the text.
Also link to the original code as a comment at the top of the new hash function (https://werxltd.com/wp/2010/05/13/javascript-implementation-of-javas-string-hashcode-method/)